kube-core/src/dynamic.rs

```diff
@@ -57,6 +71,13 @@ impl DynamicObject {
         self.metadata.namespace = Some(ns.into());
         self
     }
+
+    /// Attempt to convert this `DynamicObject` to a `Resource`
+    pub fn try_parse<K: Resource + for<'a> serde::Deserialize<'a>>(
```

Comment:
what is this strange `for<'a>` generic bound? :thinking:

Reply:
It is a [Higher-Rank trait Bound](https://doc.rust-lang.org/nomicon/hrtb.html). I had never seen this before, but the compiler suggested it to solve the life time issues. After reading the docs I think it makes sense to use here.

Reply:
Ah, ok. That makes sense. I think we've been cheating about this generally elsewhere using `DeserializeOwned`, but this is probably better :+1: